fluent/syntax/ftlstream.py

```diff
@@ -73,20 +73,31 @@ def take_char(self, f):
             return ch
         return None
 
-    def is_id_start(self):
-        if self.ch is None:
+    def is_char_id_start(self, ch=None):
+        if ch is None:
             return False
 
-        cc = ord(self.ch)
-
+        cc = ord(ch)
         return (cc >= 97 and cc <= 122) or \
-               (cc >= 65 and cc <= 90) or \
-            cc == 95
+               (cc >= 65 and cc <= 90)
+
+    def is_message_id_start(self):
+        if self.current_is('-'):
+            self.peek()
+
+        ch = self.current_peek()
+        is_id = self.is_char_id_start(ch)
+        self.reset_peek()
+        return is_id
 
     def is_number_start(self):
-        cc = ord(self.ch)
+        if self.current_is('-'):
+            self.peek()
 
-        return (cc >= 48 and cc <= 57) or cc == 45
+        cc = ord(self.current_peek()[0])
```

Comment:
Why is this taking the first char of the current peek? In other places you're not doing it.